dart/lib/src/event_processor/exception/exception_group_event_processor.dart

```diff
@@ -0,0 +1,23 @@
+import '../../event_processor.dart';
+import '../../protocol.dart';
+import '../../hint.dart';
+
+/// Group exceptions into a flat list with references to hierarchy.
+class ExceptionGroupEventProcessor implements EventProcessor {
+  @override
+  SentryEvent? apply(SentryEvent event, Hint hint) {
+    final sentryExceptions = event.exceptions ?? [];
+    if (sentryExceptions.isEmpty) {
+      return event;
+    }
+    final firstException = sentryExceptions.first;
+
+    if (sentryExceptions.length > 1 || firstException.exceptions == null) {
+      // If already a list or no child exceptions, no grouping possible/needed.
+      return event;
+    } else {
+      final grouped = firstException.flatten().reversed.toList(growable: false);
```

Comment:
We might not need to reverse the order, as the references to the parent should be enough. Then we can also revert the test expectations in the client tests.